sympy/assumptions/handlers/ntheory.py

```diff
@@ -246,3 +250,30 @@ def Basic(expr, assumptions):
                 return None
             return not _even
         return _integer
+
+
+AskCoprimeHandler = Dispatcher('AskCoprimeHandler')
+
+@AskCoprimeHandler.register(Expr, Expr)
+def Expr_Expr(x, y, *, assumptions):
+    if x == y:
+        return False
+
+    assumps = conjuncts(assumptions)
+    if Q.coprime(x,y) in assumps:
+        return True
+    if ~Q.coprime(x,y) in assumps:
+        return False
+    # detect even number or odd number
+    # other logics are not implemented yet
+    even_x, even_y = ask(Q.even(x), assumptions), ask(Q.even(y), assumptions)
+    if not even_x and not even_y:
+        return None
+    return fuzzy_nand([even_x, even_y])
+
+@AskCoprimeHandler.register(Integer, Integer)
+def Integer_Integer(x, y, *, assumptions):
+    gcdivisor = gcd(x,y)
```

Comment:
How should we be handling negative integers? Q.prime is only true for positive integers, but I'm not sure if that means the same should be the case for Q.coprime.

Reply:
Thinking about it, I think we shouldn't disallow negative integers. That keeps the definition simple (coprime(x, y) <=> gcd(x, y) == 1), and I think it's common to say things like "a rational number p/q is reduced if p and q are coprime".